uint16 id
int16 count
float32[] a
float32[] d